File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {

      int n = nums.length;
      int insertIndex = 0;
    
      for(int i = 0; i<n ; i++)
      {
        if(nums[i] != 0)
        {
            nums[insertIndex++] = nums[i];
        }
      }

      while(insertIndex<n)
      {
        nums[insertIndex++] = 0;
      }
    }
}

File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i = 0;
        for(int j = 0; j<nums.size(); j++)
        {
            if(nums[j] != 0)
            {
                //swap
                int tmp = nums[i];
                nums[i] = nums[j];
                nums[j] = tmp;
                //Increment i
                i++;
            }
        }
    }
};